internal/config/yaml_parser.go

```diff
@@ -17,10 +17,25 @@ package config
 import (
 	"fmt"
 	"os"
+	"strings"
 
 	"gopkg.in/yaml.v3"
 )
 
+func IsValidLogSeverity(severity string) bool {
+	switch severity {
+	case
+		"TRACE",
```

Comment:
can we define an go-lang enum type instead of using as string everywhere. I am assuming when you write logs, you need to refer to these strings.

Reply:
done